### DISCUSSION TITLE
AttributeError: 'Batch' object has no attribute 'xxx'

### DISCUSSION THREAD

**Discussion by mfarazi1991** [pyg-team/pytorch_geometric#3965] [Q&A] [answered]
Hi,

I am using ChebConv layer in my network. However, my data loader has another component which is a pre-computed Laplacian and its lambda_max. So when I create my data_loader object I have 

Batch(LBO_index=[2, 15139999], LBO_weight=[15139999], batch=[1226551], edge_index=[2, 13913448], lmax=[8], pos=[1226551, 3], ptr=[9], x=[1226551, 3], y=[8])

But, when I pass it to my network I get the error:

**AttributeError: 'Batch' object has no attribute 'LBO_index'**

I cannot understand why this happens. Any thoughts would be appreciated.

**Comment by rusty1s** [comment, marked as answer]
Do you have a small example to reproduce (at best as minimal as possible)?

**Reply by mfarazi1991** [reply]
Ok, I found the problem. The thing is when I use pre-computed Laplacian for a mesh, after using the methods like graclus clustering, I have to customized all the methods as they do will create new batches without my LBO_index and LBO_weights. But the problem is, as I coarsen a mesh, I no longer can update the laplacian in a new structure. This is something I cannot figure out. There is a get_laplacian at the beginning of ChebConv and replaces it with my LBO matrix but after coarsening I do not know how to update the laplacian on new set of vertices.

**Reply by rusty1s** [reply]
I see. Yeah, this attributes will likely needed to be re-computed after coarsening. You can either drop the pre-computed Laplacian for the coarsened stage and let `ChebConv` compute them for you, or you need to re-apply the procedure to obtain your `LBO_index` or `LBO_weight` on the coarsened graph.

**Reply by mfarazi1991** [reply]
Thank you so much! I believe I have to come up with approximation method of LBO on coarsen mesh.
